src/main/java/net/dv8tion/jda/internal/interactions/response/CallbackResponseImpl.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2015 Austin Keener, Michael Ritter, Florian Spieß, and the JDA contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.dv8tion.jda.internal.interactions.response;
+
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.interactions.response.CallbackResponseUnion;
+import net.dv8tion.jda.api.requests.restaction.interactions.InteractionCallbackAction;
+import net.dv8tion.jda.api.utils.data.DataObject;
+import net.dv8tion.jda.internal.interactions.InteractionHookImpl;
+
+import javax.annotation.Nonnull;
+
+public class CallbackResponseImpl implements CallbackResponseUnion
+{
+    private final InteractionCallbackAction.ResponseType type;
+
+    private Message message;
+
+    public CallbackResponseImpl(InteractionHookImpl hook, DataObject resource)
+    {
+        this.type = InteractionCallbackAction.ResponseType.fromId(resource.getInt("type", -1));
+
+        if (type.doesCallbackResponseContainMessage())
+        {
+            resource.optObject("message")
+                    .ifPresent(message -> this.message = hook.buildMessage(message));
+        }
```

Comment:
I'd recommend assigning the field outside of the lambda, then making the field final.

```suggestion
        if (type.doesCallbackResponseContainMessage())
            this.message = resource.optObject("message").map(hook::buildMessage).orElse(null);
        else
            this.message = null;
```